src/components/app.tsx

```diff
@@ -16,10 +17,9 @@ interface IState {}
 export class AppComponent extends BaseComponent<IProps, IState> {
 
   public render() {
-    const {ui} = this.stores;
     return (
       <div className="app-container">
-        <TopBarComponent />
+        {urlParams.topBar && <TopBarComponent />}
```

Comment:
I feel like there's a potential for a spaghetti of different components asking about urlParams in a just-in-time kind of way. Did you consider using the url params once to set up the initial stores, and then only referencing everything through the stores?

E.g. `showTopBar` seems like it's a natural property of the `ui` store. If we created the stores in index.tsx with `const stores = createStores({ urlParams });`, then every store could grab the properties it needs and initialize itself. (We could even leverage the nesting of the JSON to automatically pass the relevant props to the relevant stores, maybe.)

Then `AppComponent` would just check the store to chose whether to display `<TopBarComponent />`.

Another potential advantage is that `AuthoringComponent` could be displayed by `AppComponent` (instead of switching to it in index.tsx) and then it would also have easy access to all the stores, and it could pre-populate the authoring form with the current values from the store. So if authors had a long url and they just wanted to tweak one thing, they would simply add `&authoring` to the url and see the form already set up. This would also work if we ever went with LARA authoring, where the JSON is stored on the backend, and an author would expect to see the correct form for a model they've already authored.

Reply:
I hadn't considered this initially! I chose this path because it's how URL parameters work in CLUE. I liked your idea though and got most of the way through implementing it before I had some second thoughts.

I agree that `showTopBar` is information that could go in the UI model, but as far as I know, the value will never change during execution. I'm not sure how I feel about adding an extra piece of state to the model just so that we can pull values from a store instead of from the URL params.

As far as rendering right in `index.tsx` vs. in `AppComponent` - I'm not sure how things work in LARA, but at the moment it would still be easy to support adding `&authoring` to bring up a pre-loaded form, just by grabbing the current URL params. It's definitely true that this wouldn't work if we wanted to pull values out of state, but I can't think of when we'd want to do that off the top of my head.

Let me know what you think, happy to talk more next week! @kswenson, also curious if you have any CLUE-related thoughts.

Reply:
I think @sfentress is right that reading URL parameters from arbitrary points in the code sets a bad precedent. I think it would make more sense for `index.tsx` to read the URL parameters and distribute them appropriately. Whether that distribution should be through the stores is a separate question, however. In this instance it would make sense for it to be passed as a prop to `AppComponent`, in which case whether or not it's in stores becomes an implementation detail.

Reply:
I could see `showTopBar` being either in a prop or the store, but in any case I feel like many of the props will end up being things that will be in stores -- e.g. the initial mice in the backpack (users can might more), the initial tool being shown (users can change it), etc. Most of what an author will be doing will be setting state, I think, so I feel it makes sense to store it as state.

Reply:
@kswenson @sfentress Good points all around. I just added a new commit that addresses concerns about references to `url-params`. Now, all URL processing is done in `index.tsx`. I passed `showTopBar` as a prop to the `AppComponent`, but would definitely pass params to store-creation in the future for things like `initialMice`.

One question I considered was where to put default values. I opted for putting them in `url-params` rather than adding default values at the point of use, which is more or less what we do in CLUE.

There are a lot of small decisions to make here - let's hop on a call if there are any more changes you'd like to see/consider.

Reply:
Oh also - in a perfect world, I'd add an Enzyme test to make sure the App renders correctly with the new prop, but I want to get this merged sooner so that others can use it.